time broadcastTime
float32 duration
circleEntry[] array
